### DISCUSSION TITLE
How to ignore part of command. eg: tsh ssh root@**

### DISCUSSION THREAD

**Discussion by EliottGoye** [junegunn/fzf#3277] [Q&A]
Hi there,

I am trying to implement a fzf custom completion script for the `tsh` CLI that works pretty much like `ssh` and I would like to ignore the text right before the `**`

Example :

```bash
tsh ssh root@**
```

Actually the `root@` part is within the search. Is it possible to remove the user from the search so I can search from hosts ?

My completion script at the moment :

```bash
_fzf_complete_tsh() {
    ARGS="$*"
    if [[ $ARGS == 'tsh ssh'* ]]; then
        _fzf_complete --height=60% --info=inline --border --margin=1 --padding=1 -- "$@" < <( tsh ls -f names 
        )     
    else  
        eval "zle ${fzf_default_completion:-expand-or-complete}"
    fi    
}
```

**Comment by LangLangBart** [comment]
```sh
# see explanation in "man fzf"
echo "tsh ssh root@**test1\ntsh ssh root2@**test2" | fzf --delimiter @ --with-nth 2..
```
<img src="https://user-images.githubusercontent.com/92653266/236018229-fa971c37-7cc0-4036-a79e-d9c991b36031.png" width="500">

**Reply by EliottGoye** [reply]
I don't want to ignore part of the input, but part of arguments, the `root@` part.

I'm using this part of the docs :
https://github.com/junegunn/fzf#custom-fuzzy-completion

I also noticed the fzf custom completion script does not use the `--query` parameter

Example:

```bash
_fzf_complete_tsh() {
 _fzf_complete --query "toto" --height=60% --info=inline --border --margin=1 --padding=1 -- "$@" < <( tsh ls -f names )
}
```

I was thinking about getting the user `root@` from the arguments, remove it from the search with `--query=""` and re-add it with a post-process command

Here is an example `tsh ls -f names` output :

```bash
> tsh ls -f names

ansible
backup-server
dev-mariadb1
dev-mariadb2
dev-mariadb3
integration1
integration2
integration3
integration4
integration5
```

**Comment by junegunn** [comment]
> does not use the --query parameter

The completion function overrides your `--query` with `-q "${(Q)prefix}"`.

https://github.com/junegunn/fzf/blob/94999101e358385f3ca67a6ec9512f549196b802/shell/completion.zsh#L211

So, if you want to remove it, you can clear the initial `--query` with `clear-query` binding like so

```sh
--bind start:clear-query
```

**Reply by EliottGoye** [reply]
Thanks ! 
Had to upgrade my version of fzf but that did the trick !

```bash
_fzf_complete_tsh() {
 _fzf_complete --bind start:clear-query --height=60% --info=inline --border --margin=1 --padding=1 -- "$@" < <( tsh ls -f names )
}
```

Now I'm searching for a way to get the dropped initial query into a var so I can use it in a post script

```bash
# User hardcoded
_fzf_complete_tsh_post() {
  awk '{print "root@"$NF}'
}

# What I'd like to have
_fzf_complete_tsh_post() {
  awk '{print "${tsh_user}@"$NF}'
}
```

**Reply by junegunn** [reply]
I would store the query string in a temporary file and read it in the post function. e.g. `--bind 'start:execute-silent(echo {q} > /tmp/q)+clear-query'`

**Reply by EliottGoye** [reply]
Wow, the `--bind=EVENT` can do so much more than I imagined !

Here is my code atm :

```bash
_fzf_complete_tsh() {
    ARGS="$*"
    if [[ $ARGS == 'tsh ssh'* ]]; then
            _fzf_complete --bind 'start:execute-silent(echo {q} > /tmp/tsh_user)+clear-query' --height=60% --info=inline --border --margin=1 --padding=1 -- "$@" < <( tsh ls -f names )
    else  
        eval "zle ${fzf_default_completion:-expand-or-complete}"
    fi    
}

_fzf_complete_tsh_post() {
  awk -v user=$(cat /tmp/tsh_user) '{print user $NF}'
}
```

Weird thing, when the post function runs, it returns the user from the previous completion.
I've watched the file to check if there was some latency, but no. Strange behavior.
